src/layout/MasterLayout.tsx

```diff
@@ -7,13 +7,34 @@ import "../assets/styles/MasterLayout.scss"
 import { useAppContext } from '../helpers/app-store'
 import { useNavigate } from 'react-router-dom'
 import { Outlet } from 'react-router-dom'
+import axios from 'axios'
+import { toast } from 'react-toastify'
+
+
 const MasterLayout = () => {
   const { username } = useAppContext();
   const navigate = useNavigate();
+
   useEffect(() => {
     if (!username) {
```

Comment:
its better to minimize it like this in one single line
`if(!username) return navigate('...')`